sstables/sortedmap_key_index.go

```diff
@@ -0,0 +1,128 @@
+package sstables
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"golang.org/x/exp/slices"
+
+	rProto "github.com/thomasjungblut/go-sstables/recordio/proto"
+	"github.com/thomasjungblut/go-sstables/skiplist"
+	"github.com/thomasjungblut/go-sstables/sstables/proto"
+)
+
+// SortedMapIndex is keeping the entire index as a map and the ordered keys as a slice in memory
+// Use binary search to iter over the index.
+type SortedMapIndex struct {
+	index map[[20]byte]IndexVal
+	keys  [][20]byte
```

Comment:
ah I see you need this for the range scans. 
I can see two ways to deal with it: either we panic on the range methods, or you can also type embed the other index instead of reimplementing all functions